src/lib/arch/riscv/Instruction_execute.cc

```diff
@@ -0,0 +1,664 @@
+      int64_t out = signExtendW(static_cast<int32_t>(rs1 << rs2));
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SLLIW: {
+      const int32_t rs1 = operands[0].get<uint32_t>();
+      const int32_t rs2 =
+          metadata.operands[2].imm & 63;  // Only use lowest 6 bits
+      uint64_t out = signExtendW(static_cast<uint32_t>(rs1 << rs2));
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRL: {
+      const uint64_t rs1 = operands[0].get<uint64_t>();
+      const uint64_t rs2 =
+          operands[1].get<uint64_t>() & 63;  // Only use lowest 6 bits
+      uint64_t out = static_cast<uint64_t>(rs1 >> rs2);
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRLI: {
+      const uint64_t rs1 = operands[0].get<uint64_t>();
+      const uint64_t rs2 =
+          metadata.operands[2].imm & 63;  // Only use lowest 6 bits
+      uint64_t out = static_cast<uint64_t>(rs1 >> rs2);
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRLW: {
+      const uint32_t rs1 = operands[0].get<uint32_t>();
+      const uint32_t rs2 =
+          operands[1].get<uint32_t>() & 63;  // Only use lowest 6 bits
+      uint64_t out = signExtendW(static_cast<uint64_t>(rs1 >> rs2));
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRLIW: {
+      const uint32_t rs1 = operands[0].get<uint32_t>();
+      const uint32_t rs2 =
+          metadata.operands[2].imm & 63;  // Only use lowest 6 bits
+      uint64_t out = signExtendW(static_cast<uint32_t>(rs1 >> rs2));
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRA: {
+      const int64_t rs1 = operands[0].get<int64_t>();
+      const int64_t rs2 =
+          operands[1].get<int64_t>() & 63;  // Only use lowest 6 bits
+      int64_t out = static_cast<int64_t>(rs1 >> rs2);
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRAI: {
+      const int64_t rs1 = operands[0].get<int64_t>();
+      const int64_t rs2 =
+          metadata.operands[2].imm & 63;  // Only use lowest 6 bits
+      int64_t out = static_cast<int64_t>(rs1 >> rs2);
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRAW: {
+      const int32_t rs1 = operands[0].get<int32_t>();
+      const int32_t rs2 =
+          operands[1].get<int32_t>() & 63;  // Only use lowest 6 bits
+      int64_t out = static_cast<int32_t>(rs1 >> rs2);
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_SRAIW: {
+      const int32_t rs1 = operands[0].get<int32_t>();
+      const int32_t rs2 =
+          metadata.operands[2].imm & 63;  // Only use lowest 6 bits
+      int64_t out = static_cast<int32_t>(rs1 >> rs2);
+      results[0] = out;
+      break;
+    }
+    case Opcode::RISCV_ADD: {
+      const uint64_t n = operands[0].get<uint64_t>();
+      const uint64_t m = operands[1].get<uint64_t>();
+      uint64_t out = static_cast<uint64_t>(n + m);
```

Comment:
Probably don't need the static cast - but also shouldn't effect anything by including it...

Reply:
Same applies for other cases below where both operand types are the same before doing operation and isn't sign extended etc

Think needed for shifts though. Correct me if im wrong at all